scripts/config/msu/classes/enum.nut

```diff
@@ -0,0 +1,49 @@
+::MSU.Class.Enum <- class
```

Comment:
Why do you not allow removing an item?

Reply:
That could cause significant issues depending on when it was done and I can't think of any reason to do this.

Reply:
Yeah I think removing an item from an enum is a big nono

Reply:
@Both:  I do see an issue that if someone removes an entry from the Enum and some other mod tries to access it, then yeah it will blow up. That'll blow up even in the case of the table stand-ins. So yeah, I agree, don't allow removing entries.  I guess I was talking more from a usability scope point of view - otherwise even adding items to an enum is bad (from a code perspective - as enums are primarily meant to be compile time).

Regarding usability, I consider this to be a replacement for the vanilla BB method of using Tables as stand-ns for enums. There the following things are important:
1. Adding/removing items from that table.
2. Cloning the table.
3. Iterating over the table.

Therefore, in my opinion, If this enum class is to offer a better alternative to that Table stand-in method, then it should be able to do these things as well (except perhaps the removal thing).

Reply:
So we agreed to add cloning and iteration?

Reply:
Yes, from what I remember we did wanna add cloning and iteration, I will take a look at doing that.